thirdparty/clang/.clang-format

```diff
@@ -0,0 +1,203 @@
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: Never
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: Empty
+AllowShortLambdasOnASingleLine: false
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: true
+AlwaysBreakTemplateDeclarations: Yes
+BinPackArguments: false
+BinPackParameters: false
+BraceWrapping:
+  AfterCaseLabel:  true
+  AfterClass:      true
+  AfterControlStatement: true
+  AfterEnum:       true
+  AfterFunction:   true
+  AfterNamespace:  true
+  AfterObjCDeclaration: true
+  AfterStruct:     true
+  AfterUnion:      true
+  AfterExternBlock: true
+  BeforeCatch:     true
+  BeforeElse:      true
+  IndentBraces:    false
+  SplitEmptyFunction: false
+  SplitEmptyRecord: false
+  SplitEmptyNamespace: false
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Custom
+BreakBeforeInheritanceComma: false
+BreakInheritanceList: BeforeColon
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeColon
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     120
+CommentPragmas:  '^ IWYU pragma:'
+CompactNamespaces: false
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+ConstructorInitializerIndentWidth: 4
+ContinuationIndentWidth: 4
+Cpp11BracedListStyle: true
+DeriveLineEnding: true
+DerivePointerAlignment: true
+DisableFormat:   false
+ExperimentalAutoDetectBinPacking: false
+FixNamespaceComments: true
+ForEachMacros:
+  - foreach
+  - Q_FOREACH
+  - BOOST_FOREACH
+IncludeBlocks:   Regroup
+IncludeCategories:
+  # system includes, e.g. <iostream> or <unistd.h> or <sys/types.h>
+  - Regex: '^<((linux|sys|net|netinet|arpa|crypt|rpc)\/)?([A-Za-z0-9_-]+\/?)(.h(pp)?)?>'
+    Priority: 0
+    SortPriority: 0
+  # local project (or private) includes, e.g. "Access.hpp"
+  - Regex: '^"[a-zA-Z0-9_-]+(\/[a-zA-Z0-9_-]+)*\.h(pp)?"'
+    Priority: 2
+    SortPriority: 2
+  - Regex:           '^<.*OpenSpaceToolKit\/Core.*\.hpp>'
+    Priority:        3
+    SortPriority:    3
+    CaseSensitive:   false
+  - Regex:           '^<.*OpenSpaceToolKit\/IO.*\.hpp>'
+    Priority:        4
+    SortPriority:    4
+    CaseSensitive:   false
+  - Regex:           '^<.*OpenSpaceToolKit\/Mathematics.*\.hpp>'
+    Priority:        5
+    SortPriority:    5
+    CaseSensitive:   false
+  - Regex:           '^<.*OpenSpaceToolKit\/Physics.*\.hpp>'
+    Priority:        6
+    SortPriority:    6
+    CaseSensitive:   false
+  - Regex:           '^<.*OpenSpaceToolKit\/Astrodynamics.*\.hpp>'
+    Priority:        7
+    SortPriority:    7
+    CaseSensitive:   false
+  # third party headers, e.g. <boost/algorithm/string.hpp> or <opencv2/imgproc.hpp>
+  - Regex: '^<([a-zA-Z0-9_-]+\/)+[a-zA-Z0-9_-]+\.h(pp)?>'
+    Priority: 1
+    SortPriority: 1
```

Comment:
Should we move this above, just to sort by priority?

Reply:
I've put it at the bottom so that overlapping regexs are applied in that order. In this case the third party headers outside of other OSTk headers etc.